messaging/broker.go

```diff
@@ -639,6 +689,8 @@ type Topic struct {
 	index uint64 // current index
 	path  string // on-disk path
 
+	tombstone string // Where to create post-truncation tombstone.
```

Comment:
This doesn't need to be a field since it can simply be computed from the `path`. Can you make a `Topic.tombstonePath()` instead?